core/src/validator.rs

```diff
@@ -1137,6 +1141,22 @@ impl Validator {
             ))
         };
 
+        // test-validator crate may start the validator in a tokio runtime
+        // context which forces us to use the same runtime because a nested
+        // runtime will cause panic at drop.
+        // Outside test-validator crate, we always need a tokio runtime (and
+        // the respective handle) to initialize the turbine QUIC endpoint.
+        let current_runtime_handle = tokio::runtime::Handle::try_current();
+        let tpu_client_next_runtime =
+            (current_runtime_handle.is_err() && config.use_tpu_client_next).then(|| {
+                tokio::runtime::Builder::new_multi_thread()
+                    .enable_all()
+                    .worker_threads(4)
```

Comment:
are you sure you actually need 4 workers for this? You can use tokio runtime wrapper from threadmanager to measure how active these workers really are and whether you need more/less of them.

Reply:
I checked with local validator setup that 2 workers is enough for producing outflow of 100k tps.